src/sentry/api/bases/sentryapps.py

```diff
@@ -317,11 +317,19 @@ class SentryAppInstallationsBaseEndpoint(IntegrationPlatformEndpoint):
 
     def convert_args(self, request: Request, organization_slug, *args, **kwargs):
         if is_active_superuser(request):
-            organization = organization_service.get_org_by_slug(slug=organization_slug)
+            if options.get("api.id-or-slug-enabled") and str(organization_slug).isnumeric():
```

Comment:
This section is a bit confusing to read, I think we can extract the `is_active_superuser` logic like:
```Python
extra_kwargs = {}
if is_active_superuser(request):
    extra_kwargs["user_id"] = request.user.id
if options.get("api.id-or-slug-enabled") and ...:
    organization = organization_service.get_org_by_id(id=int(organization_slug), **extra_kwargs)
```